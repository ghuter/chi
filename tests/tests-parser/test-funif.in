putsCategory :: fun (age: u8) {
	if i < 4 {
		puts("Baby");
	} else if i < 10 {
		puts("Child");
	} else if i < 20 {
		puts("Teen");
	} else if i < 65 {
		puts("Adult");
	} else {
		puts("Senior");
	}
}
