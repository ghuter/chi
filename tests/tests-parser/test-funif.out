SFUN(putsCategory, age : u8, {SIF(EBINOP(LT, EMEM(i), ECSTI(4)) : ?, SCALL(puts, ECSTS(Baby)), ELSE SIF(EBINOP(LT, EMEM(i), ECSTI(10)) : ?, SCALL(puts, ECSTS(Child)), ELSE SIF(EBINOP(LT, EMEM(i), ECSTI(20)) : ?, SCALL(puts, ECSTS(Teen)), ELSE SIF(EBINOP(LT, EMEM(i), ECSTI(65)) : ?, SCALL(puts, ECSTS(Adult)), ELSE SCALL(puts, ECSTS(Senior))))))}) : void
